libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2877,7 +2876,16 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 					if (from.location() == DataLocation::Memory)
 						body = "converted := value";
 					else if (from.location() == DataLocation::CallData)
-						solUnimplemented("Conversion of calldata types not yet implemented.");
+						body = Whiskers(R"(
+							converted := <allocateMemoryArray>(<length>)
+							<copyToMemory>(value, converted, mul(<length>, <sizeInBytes>))
+						)")
+						("functionName", functionName)
+						("allocateMemoryArray", allocateMemoryArrayFunction(to))
+						("copyToMemory", copyToMemoryFunction(from.location() == DataLocation::CallData))
+						("length", from.length().str())
+						("sizeInBytes", to.baseType()->memoryDataSize().str())
+						.render();
```

Comment:
It is really unfortunate that array copying for calldata arrays are scattered into two different functions.
Here it does static arrays